### PR TITLE
ARROW-15355: [Docs] Trigger sphinx build on documentation changes

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

**Comment by kszucs** [comment]
@jorisvandenbossche the build gets triggered correctly. We could merge this one and rebase https://github.com/apache/arrow/pull/12169 to fix the actual build error.

**Comment by jorisvandenbossche** [comment]
>  We could merge this one and rebase #12169 to fix the actual build error.

Whoops, I only see this now and just merged #12169

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -32,6 +32,7 @@ on:
       - 'ci/scripts/js_build.sh'
       - 'ci/scripts/python_build.sh'
       - 'ci/scripts/r_build.sh'
+      - 'docs/**'
```
Comment (jorisvandenbossche) [inline .github/workflows/docs.yml:35]: This should only trigger the "light" job and not the "complete" job?

Comment (kszucs) [reply .github/workflows/docs.yml:35]: In order to do that we need to split the workflow file into two.

Comment (kszucs) [reply .github/workflows/docs.yml]: Split into two workflows so we trigger the light build only on pull requests and the complete docs build only on pushes.

**Review by kszucs — COMMENTED** [review]
+1

**Pull request closed without merging**